scripts/ansible/roles/windows/jenkins/tasks/slave-setup.yml

```diff
@@ -0,0 +1,192 @@
+
+    - set_fact: node_secret="{{script_output.output.lstrip('Result:') | trim}}"
+
+- name: Download slave-agent.jnlp from Jenkins Master
+  win_get_url:
+    url: "{{ jenkins_url }}/computer/{{ jenkins_agent_name }}/slave-agent.jnlp"
+    force_basic_auth: yes
+    url_username: '{{ jenkins_admin_name }}'
+    url_password: '{{ jenkins_admin_password }}'
+    dest: "{{ jenkins_agent_root_dir }}/slave-agent.jnlp"
+
+- name: Make sure that Jenkins home exists
+  win_file:
+    path: "{{ jenkins_agent_root_dir }}"
+    state: directory
+
+- name: Download agent.jar from Jenkins Master
+  win_get_url:
+    url: "{{ jenkins_url }}/jnlpJars/agent.jar"
+    force_basic_auth: yes
+    url_username: "{{ jenkins_admin_name }}"
+    url_password: "{{ jenkins_admin_password }}"
+    dest: "{{ jenkins_agent_root_dir }}/agent.jar"
+
+- name: Download all the installers
+  win_get_url:
+    url: "{{ item.value.url }}"
+    dest: "{{ item.value.dest }}"
+    timeout: 60
+  retries: 5
+  with_dict: "{{ packages }}"
+
+- name: Add all the packages to the machine PATH
+  win_path:
+    elements: "{{ item }}"
+  with_items: "{{ packages_env_path }}"
+
+- name: CMake - Install
+  win_package:
+    path: "{{ packages['cmake']['dest'] }}"
+
+- name: 7Zip - Install
+  win_package:
+    path: "{{ packages['7z']['dest'] }}"
+
+- name: Microsoft Visual Studio 2017 - Install
+  raw: "{{ packages['vs_2017']['dest'] }} --quiet --wait --add Microsoft.VisualStudio.Workload.VCTools --includeRecommended"
+
+- name: Clang 7 - Install
+  win_package: 
+    path: "{{ packages['clang7']['dest'] }}"
+    product_id: 'LLVM'
+
+- name: Git - Install
+  raw: "{{ packages.git.dest }} /SILENT"
+
+- name: OCaml - Extract
+  raw: '7z.exe x "{{ packages.ocaml.dest }}" -r -y -o"C:\Program Files"'
+
+- name: JDK8 - Install
+  win_chocolatey:
+    name: jdk8
+    state: latest
+
+- name: Windows | Check if the service wrapper exists
+  win_stat:
+    path: "{{ jenkins_agent_root_dir }}/servicewrapper.exe"
+  register: servicewrapper_info
+
+- name: Windows | Get the service wrapper
+  block:
+    - name: Windows | Download the service wrapper zip file
+      win_get_url:
+        url: "{{ servicewrapper_url }}"
+        dest: "{{ tmp_dir }}\\service-wrapper.zip"
+        timeout: 60
+      retries: 3
+
+    - name: Windows | Unzip the service-wrapper zip archive
+      win_unzip:
+        src: "{{ tmp_dir }}\\service-wrapper.zip"
+        dest: "{{ tmp_dir }}\\service-wrapper"
+
+    - name: Windows | Move the service wrapper to the expected location
+      win_copy:
+        src: "{{ tmp_dir }}\\service-wrapper\\service-wrapper.exe"
+        dest: "{{ jenkins_agent_root_dir }}\\servicewrapper.exe"
+        remote_src: yes
+
+    - name: Windows | Remove service-wrapper temporary files
+      win_file:
+        state: absent
+        path: "{{ item }}"
+      with_items:
+        - "{{ tmp_dir }}\\service-wrapper.zip"
+        - "{{ tmp_dir }}\\service-wrapper\\service-wrapper"
```

Comment:
You may delete the whole directory used when extracting.

Use
```
"{{ tmp_dir }}\\service-wrapper"
```
instead of 
```
"{{ tmp_dir }}\\service-wrapper\\service-wrapper"
```

Reply:
Changed to match recommendation.